### ISSUE TITLE
`pre-commit autoupdate` "cycles" on ansible-lint repo

### ISSUE THREAD

**Issue by pyhedgehog** [pre-commit/pre-commit#3035]
### search you tried in the issue tracker

autoupdate, latest, ansible

### describe your issue

I'm using https://github.com/ansible/ansible-lint, but they using moving tags (latest and v6) that points to same commit that actual version (v6.21.1 right now).

```console
$ pre-commit autoupdate --repo https://github.com/ansible/ansible-lint
Updating https://github.com/ansible/ansible-lint ... updating v6.21.1 -> latest.

$ pre-commit autoupdate --repo https://github.com/ansible/ansible-lint
[WARNING] The 'rev' field of repo 'https://github.com/ansible/ansible-lint' appears to be a mutable reference (moving tag / branch).  Mutable references are never updated after first install and are not supported.  See https://pre-commit.com/#using-the-latest-version-for-a-repository for more details.  Hint: `pre-commit autoupdate` often fixes this.
Updating https://github.com/ansible/ansible-lint ... already up to date.
```
### pre-commit --version

pre-commit 2.17.0

### .pre-commit-config.yaml

```yaml
repos:
-   repo: https://github.com/ansible/ansible-lint
    rev: v6.21.1
    hooks:
    -   id: ansible-lint
```

Updates to:

```yaml
repos:
-   repo: https://github.com/ansible/ansible-lint
    rev: latest
    hooks:
    -   id: ansible-lint
```

## Proposal
Let's add option to `autoupdate` to skip "mutable" tags (as defined in `WarnMutableRev`) ever if it's latest.
Not sure it this should be opt-in or opt-out flag...

**Comment by asottile** [comment]
you're using an unsupported old version of pre-commit.  it does not do this with the latest version

**Comment by pyhedgehog** [comment]
@asottile It's true, sorry. It's latest available in Ubuntu 22.04. Had to install using pipx.
